fiona/collection.py

```diff
@@ -123,13 +128,12 @@ def __init__(
             elif 'geometry' not in schema:
                 raise SchemaError("schema lacks: geometry")
             self._schema = schema
-            
-            if crs:
-                if 'init' in crs or 'proj' in crs or 'epsg' in crs.lower():
-                    self._crs = crs
-                else:
-                    raise CRSError("crs lacks init or proj parameter")
```

Comment:
@sgillies Looks like this `if/else` is the only check for an invalid `crs` parameter.  Without it any invalid `crs` is silently ignored for shapefiles and is silently changed to `EPSG:4326` for GeoJSON.  Didn't try any other formats and I'm guessing the GeoJSON change is actually happening in the OGR GeoJSON driver since that's technically the only supported CRS?

Additionally, a datasource that lacks CRS information has a `crs` property that is an empty dictionary and a `crs_wkt` property that raises a `ValueError`.  `fio info` doesn't have error handling for the `ValueError`, but it seems like that should probably be an empty string rather than an error.